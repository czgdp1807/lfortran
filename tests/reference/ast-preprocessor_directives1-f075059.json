{
    "basename": "ast-preprocessor_directives1-f075059",
    "cmd": "lfortran --indent --show-ast --no-color {infile} -o {outfile}",
    "infile": "tests/preprocessor_directives1.f90",
    "infile_hash": "ab3fb77e35a8845e8795927c7785b92c0a5b1958eb45cd16a538344f",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "ast-preprocessor_directives1-f075059.stdout",
    "stdout_hash": "dc0a6aa335d98015e1e61ea48c66f7d51d831a045125a743a900a035",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}