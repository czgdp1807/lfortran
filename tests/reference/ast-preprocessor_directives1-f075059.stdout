(TranslationUnit
    [(Module
        preprocessor_directives1
        ()
        []
        [(ImplicitNone
            []
            ()
        )]
        [(Declaration
            ()
            [(SimpleAttribute
                AttrPrivate
            )]
            []
            (TriviaNode
                []
                [(EndOfLine)
                (EndOfLine)]
            )
        )
        (Declaration
            (AttrType
                TypeCharacter
                [(len
                ()
                Star)]
                ()
                None
            )
            [(SimpleAttribute
                AttrParameter
            )]
            [(pwd_env
            []
            []
            (String
                "PWD"
            )
            Equal
            ())]
            ()
        )
        (Interface
            (InterfaceHeader)
            (TriviaNode
                []
                [(EndOfLine)
                (EndOfLine)]
            )
            [(InterfaceProc
                (Function
                    chdir
                    [(path)]
                    []
                    stat
                    (Bind
                        [C]
                        [(name
                        (String
                            "chdir"
                        ))]
                    )
                    ()
                    []
                    [(Import
                        [c_char
                        c_int]
                        ImportDefault
                        ()
                    )]
                    []
                    [(Declaration
                        (AttrType
                            TypeCharacter
                            [(kind
                            c_char
                            Value)
                            (len
                            1
                            Value)]
                            ()
                            None
                        )
                        [(AttrIntent
                            In
                        )]
                        [(path
                        [(()
                        ()
                        DimensionStar)]
                        []
                        ()
                        None
                        ())]
                        ()
                    )
                    (Declaration
                        (AttrType
                            TypeInteger
                            [(()
                            c_int
                            Value)]
                            ()
                            None
                        )
                        []
                        [(stat
                        []
                        []
                        ()
                        None
                        ())]
                        ()
                    )]
                    []
                    []
                )
            )]
        )]
        []
    )]
)
